src/app/zones/views/ZoneCreateView.vue

```diff
@@ -246,15 +248,17 @@ type ErrorState = {
 const { t } = useI18n()
 const kumaApi = useKumaApi()
 
+const NAME_REGEX = /^[a-z0-9]([-_a-z0-9]*[a-z0-9])?$/
```

Comment:
Good to know. I’ll update this.

I think at least the following message (if not also the underlying validation) on the backend side is incorrect then:

> invalid characters. Valid characters are numbers, lowercase latin letters and '-', '_' symbols.

As far as I understand, underscores should not be valid. Also, would it be good if this message actually says this should be a valid DNS name as per RFC 1035 because the rules are quite a bit more specific than "these are the valid characters"?

Reply:
Looking at https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#rfc-1035-label-names, the following should match what we need.

```
^(?![-0-9])[a-z0-9-]{1,63}$
```

(Though I’m not sure why Kubernetes needs lowercase names. That doesn't seem to be a requirement of RFC 1035)

Reply:
Because: https://datatracker.ietf.org/doc/html/rfc1035#section-2.3.1 says:

> Note that while upper and lower case letters are allowed in domain
names, no significance is attached to the case.  That is, two names with
the same spelling but different case are to be treated as if identical.

This thing of not making a difference between upper and lower case in a lot of the web (HTTP headers is another example) has caused a lot of PTSD on stupid bugs due case sensitive string comparisons. My guess is that the k8s authors just wanted to avoid future trauma by being more restrictive :)

Reply:
How about saying:

> The name is invalid. It must start with a letter, be less than 64 characters long, and only contain lowercase letters, numbers, and '-' (RFC 1035 compliant).

Reply:
Yeah, that’s better. Should we update the backend validation message, too? Or is the validation for this incorrect, too (i.e. allowing underscores)?

Reply:
I’ve changed this now and also updated the RegExp.

Reply:
There's an issue for backend